tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -916,6 +916,62 @@ public async Task GetAuthorizationRequestUrl_NoHint_NoCcsRoutingHint_TestAsync()
             }
         }
 
+        [TestMethod]
+        public void GetAuthorizationRequestUrl_WithConsumerInCreate_ReturnsConsumers()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                const string Tenant = "consumers";
+                ConfidentialClientApplicationOptions applicationOptions;
+                applicationOptions = new ConfidentialClientApplicationOptions();
+                applicationOptions.ClientId = "fakeId";
+                applicationOptions.RedirectUri = "https://example.com";
+                applicationOptions.ClientSecret = "rwerewrwe";
+
+                var confidentialClientApplicationBuilder = ConfidentialClientApplicationBuilder
+                                                                    .CreateWithApplicationOptions(applicationOptions)
+                                                                    .WithAuthority(AzureCloudInstance.AzurePublic, Tenant);
+                var confidentialClientApplication = confidentialClientApplicationBuilder.Build();
+
+                Uri authorizationRequestUrl = confidentialClientApplication
+                    .GetAuthorizationRequestUrl(new List<string> { "" })
+                    .ExecuteAsync()
+                    .ConfigureAwait(false)
+                    .GetAwaiter()
+                    .GetResult();
+
+                Assert.IsTrue(authorizationRequestUrl.Segments[1].StartsWith(Tenant));
+            }
+        }
```

Comment:
The idea is that if request changes authority to common/org/consumer, config takes precendence.

```suggestion
        public void GetAuthorizationRequestUrl_WithConsumerInCreate_ReturnsConsumers()
        {
            using (var httpManager = new MockHttpManager())
            {
                ConfidentialClientApplicationOptions applicationOptions;
                applicationOptions = new ConfidentialClientApplicationOptions();
                applicationOptions.ClientId = "fakeId";
                applicationOptions.RedirectUri = "https://example.com";
                applicationOptions.ClientSecret = "rwerewrwe";

                var confidentialClientApplicationBuilder = ConfidentialClientApplicationBuilder
                                                                    .CreateWithApplicationOptions(applicationOptions)
                var confidentialClientApplication = confidentialClientApplicationBuilder.Build();

                Uri authorizationRequestUrl = confidentialClientApplication
                    .GetAuthorizationRequestUrl(new List<string> { "" })
                    .WithAuthority(AzureCloudInstance.AzurePublic, Constants.ConsumerTenant);
                    .ExecuteAsync()
                    .ConfigureAwait(false)
                    .GetAwaiter()
                    .GetResult();

                Assert.IsTrue(authorizationRequestUrl.Segments[1].StartsWith(Constants.CommonTenant));
            }
        }
```